.github/workflows/ci.yml

```diff
@@ -0,0 +1,20 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
NIT: We want it to run only on merge ? Not on PR's it self.

Reply:
`pull_request:` means it'll run on all pull-request events. `branches: - master` limits it to run only on `master` branch.

The result is the union of these two. This is to ensure we don't get doubled runs because for PR branches both `pull_request` and `branches` events are fired. See trino's ci.yml for example.